python/ray/parameter.py

```diff
@@ -249,6 +249,55 @@ def update_if_absent(self, **kwargs):
 
         self._check_usage()
 
+    def update_pre_selected_port(self):
+        """Update the pre-selected port information
+
+        Returns:
+            The dictionary mapping of component -> ports.
+        """
+        # Create a dictionary of the component -> port mapping.
+        pre_selected_ports = {
+            "gcs": self._wrap_port(self.redis_port),
+            "object_manager": self._wrap_port(self.object_manager_port),
+            "node_manager": self._wrap_port(self.node_manager_port),
+            "gcs_server": self._wrap_port(self.gcs_server_port),
+            "client_server": self._wrap_port(self.ray_client_server_port),
+            "dashboard": self._wrap_port(self.dashboard_port),
+            "dashboard_agent": self._wrap_port(self.metrics_agent_port),
+            "metrics_export": self._wrap_port(self.metrics_export_port),
+        }
+        redis_shard_ports = self.redis_shard_ports
+        if redis_shard_ports is None:
+            redis_shard_ports = []
+        pre_selected_ports["redis_shards"] = redis_shard_ports
+        if self.worker_port_list is None:
+            if (self.min_worker_port is not None
+                    and self.max_worker_port is not None):
+                pre_selected_ports["worker_ports"] = list(
+                    range(self.min_worker_port, self.max_worker_port + 1))
+            else:
+                # The dict is not updated when it requires random ports.
+                pre_selected_ports["worker_ports"] = []
+        else:
+            pre_selected_ports["worker_ports"] = (
+                self.worker_port_list.split(","))
+
+        # Update the pre selected port set.
+        self.pre_selected_ports_set = set()
+        self.pre_selected_ports = pre_selected_ports
```

Comment:
is this variable actually used?

Reply:
No. I was originally planning to expose it to users, but there were some tricky parts to do that. Let me remove it.